list/list.js

```diff
@@ -652,6 +652,297 @@ each({
+	 *   return todos;
+	 * }, {});
+	 * todosAsOneObject //-> { dishes: false, lawn: true }
+	 * ```
+	 *
+	 * @param {function(item, index, list)} callback A function to call with each element of the DefineList.
+	 * The four parameters that callback gets passed are:
+	 * 		- current - the current aggregate value of reducing over the list -- the initial value if the first iteration
+	 *    - item - the element at index.
+	 *    - index - the current element of the list.
+	 *    - list - the DefineList the elements are coming from.
+	 *
+	 * The return value of `callback` is passed to the next iteration as the first argument, and returned from 
+	 * `reduce` if the last iteration.
+	 *
+	 * @param {*} [initialValue] The initial value to use as `current` in the first iteration
+	 * @param {Object} [thisArg] The object to use as `this` inside the callback.
+	 * @return {*} The result of the final call of `callback` on the list.
+	 * @body
+	 *
+	 */
+	"reduce": 4,
+	/**
+	 * @function can-define/list/list.prototype.reduceRight reduceRight
+	 * @description Map the values in this list to a single value from right to left
+	 *
+	 * @signature `list.reduceRight(callback, initialValue, [, thisArg])`
+	 *
+	 * Loops through the values of the list in reverse order, calling `callback` for each one until the list
+	 * ends.  The return value of `callback` is passed to the next iteration as the first argument, 
+	 * and finally returned by `reduce`.
+	 *
+	 * ```js
+	 * var todos = new DefineList([
+	 *   {name: "dishes", complete: false},
+	 *   {name: "lawn", complete: true}
+	 * ]);
+	 * var todosAsOneObject = todos.reduce(function(todos, todo){
+	 *   todos[todo.name] = todo.complete;
+	 *   return todos;
+	 * }, {});
+	 * todosAsOneObject //-> { dishes: false, lawn: true }
+	 * ```
+	 *
+	 * @param {function(item, index, list)} callback A function to call with each element of the DefineList.
+	 * The four parameters that callback gets passed are:
+	 * 		- current - the current aggregate value of reducing over the list -- the initial value if the first iteration
+	 *    - item - the element at index.
+	 *    - index - the current element of the list.
+	 *    - list - the DefineList the elements are coming from.
+	 *
+	 * The return value of `callback` is passed to the next iteration as the first argument, and returned from 
+	 * `reduce` if the last iteration.
+	 *
+	 * @param {*} [initialValue] The initial value to use as `current` in the first iteration
+	 * @param {Object} [thisArg] The object to use as `this` inside the callback.
+	 * @return {*} The result of the final call of `callback` on the list.
+	 * @body
+	 *
+	 */
+	"reduceRight": 4,
+	/**
+	 * @function can-define/list/list.prototype.every every
+	 *
+	 * Return true if every item in a list matches a predicate.
+	 *
+	 * @signature `list.every( callback [,thisArg] )`
+	 *
+	 * Filters `list` based on the return values of `callback`.  If `callback` returns truthy for every element in
```

Comment:
Maybe remove the `Filters` line... or change it to something like what MDN says:

> tests whether all elements in the array pass the test implemented by the provided function.